sample_apps/simple_app/example_key_rotation.cc

```diff
@@ -0,0 +1,190 @@
+DEFINE_bool(print_all, false,  "verbose");
+DEFINE_string(operation, "", "operation");
+
+DEFINE_string(policy_host, "localhost", "address for policy server");
+DEFINE_int32(policy_port, 8123, "port for policy server");
+DEFINE_string(data_dir, "./app1_data/", "directory for application data");
+
+DEFINE_string(policy_store_file, "store.bin", "policy store file name");
+DEFINE_string(platform_file_name, "platform_file.bin", "platform certificate");
+DEFINE_string(platform_attest_endorsement, "platform_attest_endorsement.bin", "platform endorsement of attest key");
+DEFINE_string(attest_key_file, "attest_key_file.bin", "attest key");
+DEFINE_string(measurement_file, "example_app.measurement", "measurement");
+
+
+// The test demonstrates key rotation of the public key
+// For an example of rotating keys for Protect_Blob, see the certifier_tests.
+
+#include "policy_key.cc"
+cc_trust_data* app_trust_data = nullptr;
+
+// -----------------------------------------------------------------------------------------
+
+int main(int an, char** av) {
+  gflags::ParseCommandLineFlags(&an, &av, true);
+  an = 1;
+  ::testing::InitGoogleTest(&an, av);
+
+  if (FLAGS_operation == "") {
+    printf("example_key_rotation.exe --print_all=true|false --operation=op --policy_host=policy-host-address --policy_port=policy-host-port\n");
+    printf("\t --data_dir=-directory-for-app-data\n");
+    printf("\t --policy_cert_file=self-signed-policy-cert-file-name --policy_store_file=policy-store-file-name\n");
+    return 0;
+  }
+
+  SSL_library_init();
+  string enclave_type("simulated-enclave");
+
+  string store_file(FLAGS_data_dir);
+  store_file.append(FLAGS_policy_store_file);
+  string purpose("authentication");
+
+  app_trust_data = new cc_trust_data(enclave_type, purpose, store_file);
+  if (app_trust_data == nullptr) {
+    printf("couldn't initialize trust object\n");
+    return 1;
+  }
+
+  // Init policy key info
+  if (!app_trust_data->init_policy_key(initialized_cert_size, initialized_cert)) {
+    printf("Can't init policy key\n");
+    return 1;
+  }
+
+  // Init simulated enclave
+  string attest_key_file_name(FLAGS_data_dir);
+  attest_key_file_name.append(FLAGS_attest_key_file);
+  string platform_attest_file_name(FLAGS_data_dir);
+  platform_attest_file_name.append(FLAGS_platform_attest_endorsement);
+  string measurement_file_name(FLAGS_data_dir);
+  measurement_file_name.append(FLAGS_measurement_file);
+  string attest_endorsement_file_name(FLAGS_data_dir);
+  attest_endorsement_file_name.append(FLAGS_platform_attest_endorsement);
+
+  if (!app_trust_data->initialize_simulated_enclave_data(attest_key_file_name,
+      measurement_file_name, attest_endorsement_file_name)) {
+    printf("Can't init simulated enclave\n");
+    return 1;
+  }
+
+  int ret = 0;
+  if (FLAGS_operation == "warm-restart") {
+    if (!app_trust_data->warm_restart()) {
+      printf("warm-restart failed\n");
```

Comment:
Minor: If you can change hard-coded `"warm-restart"` with `FLAGS_operation.c_str()` or something equivalent, it will bind the term in the printf() message tighter to what's being checked on L107.